lib/jnpr/junos/facts/get_software_information.py

```diff
@@ -84,11 +93,22 @@ def get_facts(device):
         if re_version is None:
             # For < 15.1, get version from the "junos" package.
             try:
-                re_pkg_info = re_sw_info.xpath(
+                junos_pkg_info = re_sw_info.xpath(
                     './package-information[name="junos"]/comment'
-                )[0].text
+                )
+                if len(junos_pkg_info):
+                    re_pkg_info = junos_pkg_info[0].text
+                else:
```

Comment:
Lines 96 - 101 can be simplified to:
```
re_pkg_info = re_sw_info.findtext('./package-information[name="junos"]/comment')
if re_pkg_info is not None:
```

Sorry, I know this is related more cleanup of my original code which used the `xpath()` method than it is related to your changes.